specs/binding-kafka.spec/src/main/java/io/aklivity/zilla/specs/binding/kafka/internal/KafkaFunctions.java

```diff
@@ -998,6 +1026,523 @@ public byte[] build()
+                        KafkaBeginExFW.FIELD_OFFSET_REQUEST + KafkaResponseBeginExFW.FIELD_OFFSET_ALTER_CONFIGS,
+                        writeBuffer.capacity());
+                }
+
+                public KafkaResponseAlterConfigsBeginExBuilder throttle(
+                    int throttle)
+                {
+                    alterConfigsBeginExRW.throttle(throttle);
+                    return this;
+                }
+
+                public KafkaResourceBuilder resource()
+                {
+                    return new KafkaResourceBuilder();
+                }
+
+                public KafkaBeginExBuilder build()
+                {
+                    final KafkaResponseAlterConfigsBeginExFW responseBeginEx = alterConfigsBeginExRW.build();
+                    beginExRO.wrap(writeBuffer, 0, responseBeginEx.limit());
+                    return KafkaBeginExBuilder.this;
+                }
+
+                public final class KafkaResourceBuilder
+                {
+                    private final MutableDirectBuffer resourceBuffer = new UnsafeBuffer(new byte[1024 * 8]);
+                    private final KafkaResourceStatusFW.Builder resourceRW = new KafkaResourceStatusFW.Builder();
+
+                    KafkaResourceBuilder()
+                    {
+                        resourceRW.wrap(resourceBuffer, 0, resourceBuffer.capacity());
+                    }
+
+                    public KafkaResourceBuilder error(
+                        short error)
+                    {
+                        resourceRW.error(error);
+                        return this;
+                    }
+
+                    public KafkaResourceBuilder message(
+                        String message)
+                    {
+                        resourceRW.message(message);
+                        return this;
+                    }
+
+                    public KafkaResourceBuilder type(
+                        String type)
+                    {
+                        resourceRW.type(t -> t.set(KafkaResourceType.valueOf(type)));
+                        return this;
+                    }
+
+                    public KafkaResourceBuilder name(
+                        String name)
+                    {
+                        resourceRW.name(name);
+                        return this;
+                    }
+
+                    public KafkaResponseAlterConfigsBeginExBuilder build()
+                    {
+                        KafkaResourceStatusFW resource = resourceRW.build();
+                        alterConfigsBeginExRW.resourcesItem(r -> r
+                            .error(resource.error())
+                            .message(resource.message())
+                            .type(resource.type())
+                            .name(resource.name()));
+
+                        return KafkaResponseAlterConfigsBeginExBuilder.this;
+                    }
+                }
+            }
+        }
+
```

Comment:
`KafkaBeginExBuilder` looks good.
Missing `*BeginExMatcherBuilder` for requests and responses.